pandas/tests/arrays/test_datetimelike.py

```diff
@@ -302,11 +303,26 @@ def test_searchsorted_castable_strings(self, arr1d, box):
         expected = np.array([1, 2], dtype=np.intp)
         tm.assert_numpy_array_equal(result, expected)
 
-        with pytest.raises(TypeError):
-            arr.searchsorted("foo")
-
-        with pytest.raises(TypeError):
-            arr.searchsorted([str(arr[1]), "baz"])
+        string = "foo"
+        with pytest.raises(
+            TypeError,
+            match=(
+                f"value should be a '{arr1d._scalar_type.__name__}', 'NaT', "
+                f"or array of those. Got '{type(string).__name__}' instead."
+            ),
+        ):
+            arr.searchsorted(string)
```

Comment:
can this (and line 325) be kept as they originally were?

Reply:
@MarcoGorelli  I see the point to keep it as original here.  For line325, is it because `extract_array()` may change later and break the test? And in contract, `StringArray` (type name) is less likely to change.

Reply:
It's not too important admittedly, it's just to minimise the amount of logic in tests as any assignment / substitution / branch has the potential to introduce bugs. When we write source code we want to write helper functions and not repeat ourselves and all of that, but for tests repetition is more acceptable - see https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

Reply:
Got it. The article is very helpful. Thanks a lot!